app/date-select/controller.js

```diff
@@ -0,0 +1,34 @@
+const FormWizardController = require('../../common/controllers/form-wizard')
+const { dateRegex } = require('../../common/helpers/url')
+
+const DateRegExp = new RegExp('/' + dateRegex)
+
+class DateSelectController extends FormWizardController {
+  get(req, res, next) {
+    if (req.query.referrer) {
+      req.sessionModel.reset()
+      req.sessionModel.set('referrer', req.query.referrer)
```

Comment:
You could maybe even use `forwardQuery` and then just always pull it from the querystring rather than needing to pull it from the session.

Reply:
Superceded